app/gui/src/controller/graph.rs

```diff
@@ -632,30 +553,46 @@ impl Handle {
         })
     }
 
-    /// Obtains information for connection's destination endpoint.
-    pub fn destination_info(
+    /// Obtains information for new connection's target endpoint.
+    pub fn target_info(
         &self,
         connection: &Connection,
         context: &impl SpanTreeContext,
     ) -> FallibleResult<EndpointInfo> {
-        let destination_node = self.node_info(connection.destination.node)?;
-        let target_node_ast = destination_node.expression();
-        EndpointInfo::new(&connection.destination, &target_node_ast, context)
+        let target_node = self.node_info(connection.target.node)?;
+        let target_node_ast = target_node.expression();
+        EndpointInfo::new(&connection.target, target_node_ast, context)
     }
 
-    /// Obtains information about connection's source endpoint.
+    /// Obtains information about new connection's source endpoint.
     pub fn source_info(
         &self,
         connection: &Connection,
         context: &impl SpanTreeContext,
     ) -> FallibleResult<EndpointInfo> {
-        let source_node = self.node_info(connection.source.node)?;
-        if let Some(pat) = source_node.pattern() {
-            EndpointInfo::new(&connection.source, pat, context)
-        } else {
-            // For subports we would not have any idea what pattern to introduce. So we fail.
-            Err(NoPatternOnNode { node: connection.source.node }.into())
+        let mut source = connection.source;
+
+        let use_whole_pattern = source.port == PortId::Root;
+        let pattern = match use_whole_pattern {
+            true => self.introduce_pattern_if_missing(connection.source.node)?,
+            false => {
+                let source_node = self.node_info(connection.source.node)?;
+                // For subports we would not have any idea what pattern to introduce. So we fail.
+                source_node
+                    .pattern()
+                    .ok_or(NoPatternOnNode { node: connection.source.node })?
+                    .clone()
+            }
+        };
```

Comment:
What is the advantage of matching bool over a single `if`? The latter is quicker to read IMO.

Reply:
I think it was more concise in this form at some point. But now it doesn't really make sense anymore.